includes/settings/tools.php

```diff
@@ -181,8 +186,17 @@ public function __construct() {
 		add_action( 'admin_post_elementor_rollback', [ $this, 'post_elementor_rollback' ] );
 	}
 
-	private function get_rollback_versions() {
+	public function get_rollback_versions() {
+
 		$rollback_versions = get_transient( 'elementor_rollback_versions_' . ELEMENTOR_VERSION );
+
+		$plugin_slug = basename( ELEMENTOR__FILE__, '.php' );
+		$is_cloud = strpos( ELEMENTOR_VERSION, '-cloud' ) !== false;
+
+		if ( $is_cloud ) {
+			$rollback_versions = apply_filters( 'elementor/settings/tools/rollback/versions', $rollback_versions, $plugin_slug );
+		}
```

Comment:
Why all this check is needed? why not only put the filter?

Reply:
There is no point in running code that does not need to be run. 

Is there a reason to run this hook outside the cloud? Cloud websites are the only ones that use this. When we don't need hook, why take the risk of activating him?